pulp_smash/pulp_smash_internals.ipynb

```diff
@@ -0,0 +1,989 @@
+    "The configuration (via `pulp-smash settings create`) is  important. But that comes after installing. And the instructions are the same regardless of how Smash is installed - user or dev.\n",
+    "\n",
+    "Settings file is split into a **pulp** and **systems** section because Pulp can be installed across separate hosts. The application-wide settings are placed into the **pulp** section, and the per-host settings are placed into the **systems** section. We don't currently perform multi-host testing of Pulp, but Smash is designed with this use case in mind, and this is one example of that.\n",
+    "\n",
+    "Perhaps a more precise definition of **systems** section is to think as **hosts**.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Good to know\n",
+    "\n",
+    "In the simplest case, Pulp Smash's configuration file resides at\n",
+    "    ``~/.config/pulp_smash/settings.json``. However, there are several ways to\n",
+    "    alter this path. Pulp Smash obeys the `XDG Base Directory Specification`_.\n",
+    "    In addition, Pulp Smash responds to the ``PULP_SMASH_CONFIG_FILE``\n",
+    "    environment variable. This variable is a relative path, and it defaults to\n",
+    "    ``settings.json``.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running your first test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "s\r\n",
+      "----------------------------------------------------------------------\r\n",
+      "Ran 0 tests in 0.000s\r\n",
+      "\r\n",
+      "OK (skipped=1)\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!python -m unittest pulp_smash.tests.pulp2.platform.api_v2.test_login"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Warning Explanation**\n",
+    "\n",
+    "If the configuration file states that HTTPS connections should not be verified, then InsecureRequestWarnings will be emitted. Smash is smart enough to suppress these warnings. But the unittest test runner overrides Smash and causes all warnings to be emitted anyway.\n",
+    "\n",
+    "\n",
+    "[Warning](https://github.com/PulpQE/pulp-smash/blob/2c5445972ff8c13cf32632f58b10b787ce2239cd/pulp_smash/tests/__init__.py#L43)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running all the tests "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All tests can be run by the following command. This action can take a while.\n",
+    "\n",
+    "``` bash\n",
+    "    python -m unittest discover pulp_smash.tests\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Installing Pulp-Smash in Development Mode\n",
```

Comment:
Maybe consider moving this section up with the User Install section? After having gone through pulp-smash settings create and running tests, this section appears out of place.